### ISSUE TITLE
Typed Dataclass instance loses type hints on import

### ISSUE THREAD

**Issue by lovetox** [microsoft/pyright#2590]
**Describe the bug**
When importing a typed variable which is a instance of a Dataclass to another module, its suddenly unknown.

**To Reproduce**
```python
from dataclasses import dataclass

@dataclass(frozen=True)
class _ErrorCondition:
    ABORTED: str = 'aborted'

ErrorCondition = _ErrorCondition()
```
then later in another module

```python
from const import ErrorCondition

x = ErrorCondition.ABORTED
```

UnknownMemberType is raised

**Expected behavior**
Members should be known

**Env**
pyright 1.1.188

SublimeText via LSP-pyright

**Comment by erictraut** [comment]
I'm not able to repro the problem as specified. There must be something more involved in the repro steps.

If you use the "Go To Declaration" feature (or whatever it's called in sublime) for the symbol `ErrorCondition` within the second module, where does it take you?

**Comment by lovetox** [comment]
Hm to the sitepackage dir where the lib is installed, this is probably because of the import resolution problem from #2587 
if you take my testproject form there, add the code from this issue, install the package you should see the error

So it pulls hints from the installed version, but its typed and fine, still unknown member type

**Comment by lovetox** [comment]
I can confirm, the error does not happen when the package is not installed

**Comment by erictraut** [comment]
OK, that explains it then. I'm running with the bug fix in place for #2587. So this problem is really a duplicate of the other problem. I'm going to close this one. If it turns out that our assumption is wrong and this isn't related to the other one, we can reopen.
